py/torch_tensorrt/fx/utils.py

```diff
@@ -37,10 +43,32 @@ def torch_dtype_to_trt(dtype: torch.dtype) -> TRTDataType:
         return trt.int8
     elif dtype == torch.int32:
         return trt.int32
+    elif dtype == torch.int64:
+        if truncate_long_and_double:
+            _LOGGER.warn(
+                "Detected Int64 Input, Casting to Int32 for TRT Engine Compatibility"
+            )
+            return trt.int32
+        else:
+            raise TypeError(
+                "Detected Int64 Input which is not supported by tensorrt, enable compilation"
+                + "option truncate_long_and_double=True to cast input to Int32 for TRT Engine"
+            )
```

Comment:
Similarly to the TorchScript path, allow the `truncate_long_and_double` argument to automatically cast inputs as needed by TRT Engines, while informing the user. This is primarily helpful for intermediate inputs (not user-provided), which happen to be long-type tensors (such as indices for embeddings).